Gemfile

```diff
@@ -19,6 +19,7 @@ gem "ahoy_matey", "~> 3.2" # Tracking analytics for Rails
 gem "ancestry", "~> 3.2" # Ancestry allows the records of a ActiveRecord model to be organized in a tree structure
 gem "blazer", "~> 2.4.2" # Allows admins to query data
 gem "bootsnap", ">= 1.1.0", require: false # Boot large ruby/rails apps faster
+gem "buffer", "~> 0.1"
```

Comment:
I'm wary to re-add a gem that we don't need (and that Ruby 2.7.2 works fine without) without understanding what the actual problem is.

Reply:
Yes, agree - I think I started on this path without fulling understanding what had been done in your branch.

I can say _what_ the problem is and where - I tried recompiling ruby with the portion I suspected removed but still saw the same behavior - it's definitely a loop (probably because the hash class method lookup table got confused) - I don't know how to distill it into something smaller.

I'm closing this PR and will let you know if I have anything smarter to say later this week.

In the meantime, I too can reproduce this issue on linux (gcc 10, rbenv built ruby 3.0.0 and ruby 3.0.1, and built outside of rbenv (from the ruby source tarball) so it's a real issue.